File: app/src/main/java/com/diploma/bank/MainMenu/MainHomeListAdapter.java
package com.diploma.bank.MainMenu;

import android.provider.ContactsContract;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.CardView;
import android.view.ViewGroup;
import android.view.LayoutInflater;
import android.widget.ImageView;
import android.widget.TextView;
import android.graphics.drawable.Drawable;
import android.support.v4.content.ContextCompat;
import android.view.View;

import com.diploma.bank.R;

class MainHomeListAdapter extends RecyclerView.Adapter<MainHomeListAdapter.ViewHolder>{

    private int[] id_card;
    private String[] card_name;
    private double[] balance;
    private int[] imageIds;

    private int[] id_deposit;
    private String[] deposit_name;
    private double[] deposit_balance;
    private int[] deposit_id_image;

    private int[] id_bank_account;
    private String[] bank_account_name;
    private double[] bank_account_balance;
    private int[] bank_account_id_image;

    private Listener listener_main_home_list;
    interface Listener {
        void onClick(int position);
    }

    public void setListener(Listener listener_main_home_list){
        this.listener_main_home_list = listener_main_home_list;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private CardView cardView;
        public ViewHolder(CardView v) {
            super(v);
            cardView = v;
        }
    }

    public MainHomeListAdapter(int[] id_card, String[] card_name, double[] balance, int[] imageIds,
                               int[] id_deposit, String[] deposit_name, double[] deposit_balance, int[] deposit_id_image){
        this.id_card = id_card;
        this.card_name = card_name;
        this.balance = balance;
        this.imageIds = imageIds;

        this.id_deposit = id_deposit;
        this.deposit_name = deposit_name;
        this.deposit_balance = deposit_balance;
        this.deposit_id_image = deposit_id_image;
    }

    @Override
    public int getItemCount(){
        return (id_card.length + id_deposit.length);
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType){
        CardView cv = (CardView) LayoutInflater.from(parent.getContext())
                .inflate(R.layout.cardview_main_home_list, parent, false);
        return new ViewHolder(cv);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, final int position){
        CardView cardView_MainMenu = holder.cardView;

        id_bank_account = new int[id_card.length + id_deposit.length];
        bank_account_name = new String[card_name.length + deposit_name.length];
        bank_account_balance = new double[balance.length + deposit_balance.length];
        bank_account_id_image = new int[imageIds.length + deposit_id_image.length];

        for (int i = 0; i < id_card.length; i++){
            id_bank_account[i] = id_card[i];
            bank_account_name[i] = card_name[i];
            bank_account_balance[i] = balance[i];
            bank_account_id_image[i] = imageIds[i];
        }

        for (int i = 0; i < id_deposit.length; i++){
            id_bank_account[i + id_card.length] = id_deposit[i];
            bank_account_name[i + card_name.length] = deposit_name[i];
            bank_account_balance[i + balance.length] = deposit_balance[i];
            bank_account_id_image[i + imageIds.length] = deposit_id_image[i];
        }

        ImageView imageView = (ImageView)cardView_MainMenu.findViewById(R.id.card_view_card_image);
        Drawable drawable = ContextCompat.getDrawable(cardView_MainMenu.getContext(), bank_account_id_image[position]);
        imageView.setImageDrawable(drawable);
        imageView.setContentDescription(bank_account_name[position]);

        TextView cardName_textView = (TextView)cardView_MainMenu.findViewById(R.id.card_view_card_name);
        cardName_textView.setText(bank_account_name[position]);

        TextView cardBalance_textView = (TextView)cardView_MainMenu.findViewById(R.id.card_view_card_balance);
        TextView cardBalanceSmall_textView = (TextView)cardView_MainMenu.findViewById(R.id.card_view_card_balance_small);

        long balance_temp = (long) Math.floor(bank_account_balance[position]);
        cardBalance_textView.setText(String.format("%, d", balance_temp));

        balance_temp = Math.round((bank_account_balance[position] % 1) * 100);
        String balance_str = ",";
        balance_str += Long.toString(balance_temp);
        cardBalanceSmall_textView.setText(balance_str);

        cardView_MainMenu.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (listener_main_home_list != null) {
                    listener_main_home_list.onClick(position);
                }
            }
        });
    }
}

File: app/src/main/java/com/vmbeshenov/bank/MainMenu/MainHomeListAdapter.kt
package com.vmbeshenov.bank.MainMenu

import android.support.v4.content.ContextCompat
import android.support.v7.widget.CardView
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.vmbeshenov.bank.R
import kotlin.math.floor
import kotlin.math.roundToInt

internal class MainHomeListAdapter(private val id_card: IntArray, private val card_name: Array<String?>, private val balance: DoubleArray, private val imageIds: IntArray,
                                   private val id_deposit: IntArray, private val deposit_name: Array<String?>, private val deposit_balance: DoubleArray, private val deposit_id_image: IntArray) : RecyclerView.Adapter<MainHomeListAdapter.ViewHolder>() {
    private lateinit var idBankAccount: IntArray
    private lateinit var bankAccountName: Array<String?>
    private lateinit var bankAccountBalance: DoubleArray
    private lateinit var bankAccountIdImage: IntArray
    private var listenerMainHomeList: Listener? = null

    internal interface Listener {
        fun onClick(position: Int)
    }

    fun setListener(listener_main_home_list: Listener?) {
        this.listenerMainHomeList = listener_main_home_list
    }

    class ViewHolder(val cardView: CardView) : RecyclerView.ViewHolder(cardView)

    override fun getItemCount(): Int {
        return id_card.size + id_deposit.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val cv = LayoutInflater.from(parent.context)
                .inflate(R.layout.cardview_main_home_list, parent, false) as CardView
        return ViewHolder(cv)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val cardViewMainMenu = holder.cardView
        idBankAccount = IntArray(id_card.size + id_deposit.size)
        bankAccountName = arrayOfNulls(card_name.size + deposit_name.size)
        bankAccountBalance = DoubleArray(balance.size + deposit_balance.size)
        bankAccountIdImage = IntArray(imageIds.size + deposit_id_image.size)
        for (i in id_card.indices) {
            idBankAccount[i] = id_card[i]
            bankAccountName[i] = card_name[i]
            bankAccountBalance[i] = balance[i]
            bankAccountIdImage[i] = imageIds[i]
        }
        for (i in id_deposit.indices) {
            idBankAccount[i + id_card.size] = id_deposit[i]
            bankAccountName[i + card_name.size] = deposit_name[i]
            bankAccountBalance[i + balance.size] = deposit_balance[i]
            bankAccountIdImage[i + imageIds.size] = deposit_id_image[i]
        }
        val imageView = cardViewMainMenu.findViewById<View>(R.id.card_view_card_image) as ImageView
        val drawable = ContextCompat.getDrawable(cardViewMainMenu.context, bankAccountIdImage[position])
        imageView.setImageDrawable(drawable)
        imageView.contentDescription = bankAccountName[position]
        val cardNameTextView = cardViewMainMenu.findViewById<View>(R.id.card_view_card_name) as TextView
        cardNameTextView.text = bankAccountName[position]
        val cardBalanceTextView = cardViewMainMenu.findViewById<View>(R.id.card_view_card_balance) as TextView
        val cardBalanceSmallTextView = cardViewMainMenu.findViewById<View>(R.id.card_view_card_balance_small) as TextView
        var balanceTemp = floor(bankAccountBalance[position]).toLong()
        cardBalanceTextView.text = String.format("%, d", balanceTemp)
        balanceTemp = (bankAccountBalance[position] % 1 * 100).roundToInt().toLong()
        var balanceStr = ","
        balanceStr += balanceTemp.toString()
        cardBalanceSmallTextView.text = balanceStr
        cardViewMainMenu.setOnClickListener {
            if (listenerMainHomeList != null) {
                listenerMainHomeList!!.onClick(position)
            }
        }
    }
}